kii_thing_if.c

```diff
@@ -634,82 +688,28 @@ static kii_bool_t prv_onboard_with_vendor_thing_id(
         M_KII_LOG(kii->kii_core.logger_cb("fail to start api call.\n"));
     }
 
-    if (kii_api_call_append_body(kii, "{\"vendorThingID\":\"",
-                    CONST_STRLEN("{\"vendorThingID\":\"")) != 0) {
+    // Open to write JSON object.
+    if (kii_api_call_append_body(kii, "{", CONST_STRLEN("{")) != 0) {
         M_KII_LOG(kii->kii_core.logger_cb("request size overflowed.\n"));
         return KII_FALSE;
     }
-    if (kii_api_call_append_body(kii, vendor_thing_id,
-                    strlen(vendor_thing_id)) != 0) {
+    // Append key value pairs.
+    if (prv_append_key_value_string(
+            kii, "vendorThingID", vendor_thing_id, NOT_SUCCESSOR) != 0 ||
+            prv_append_key_value_string(
+                kii, "thingPassword", password, IS_SUCCESSOR) != 0 ||
+            prv_append_key_value_string_optional(
+                kii, "thingType", thing_type, IS_SUCCESSOR) != 0 ||
+            prv_append_key_value_string_optional(
```

Comment:
"thingProperties" is not string, it is object, so `prv_append_key_value_string_optional` should not be used.